### DISCUSSION TITLE
Editable component should re-render on defaultValue changes

### DISCUSSION THREAD

**Discussion by user72356** [chakra-ui/chakra-ui#7085] [Ideas]
This is for the `Editable` component. I'm using a realtime system and DB changes are automatically pulled into the React interface as they occur.

How can I bind a variable to the value shown by the `EditablePreview` component? At this time the `defaultValue` prop seems like the perfect place but it's only being used one time when mounting the component. Subsequent changes are not taken into account. 

I can't use the `value` prop either since it's used to save input state while in edit mode. If I bind to it, the input value could change while the user is in the middle of making an edit himself.

I guess the solution would be if the `Editable` component should re-render on `defaultValue` changes?

**Comment by Nikavoguialy** [comment]


**Reply by Nikavoguialy** [reply]
thanks

**Comment by jrnxf** [comment]
Could we get an update on this? Ran into this problem today and it's been a frustrating one to try and create a hack for.

**Comment by mahmoudilyan** [comment]
Try using a variable using useState hook like the code below 

```  
const { color, onColorChange, onDelete, onRename } = props

  const [colorTitle, setColorTitle] = useState(color.title)

  useEffect(() => {
    setColorTitle(color.title)
  }, [color])

  return (
    <Box display="flex" alignItems="center" mb='4' justifyContent={'space-between'}>

      <Editable
        value={colorTitle}
        width='xl'
        mb='0'
        mr='4'
        alignSelf='center'
        onSubmit={(nextTitle) => {
          onRename(color.title, nextTitle)
        }}
        onChange={(nextValue) => {
          setColorTitle(nextValue)
        }}
      >
        <EditablePreview />
        <EditableInput />
      </Editable>
      ```

**Comment by samotleriche** [comment]
Are there any updates on this? I'm also experiencing this.

**Reply by og3og** [reply]
I was able to solve this by adding a unique key element to `Editable`

```
<Editable key={data.id}>
````
